presto-hive/src/test/java/com/facebook/presto/hive/parquet/write/TestDataWritableWriter.java

```diff
@@ -368,7 +371,13 @@ private void writePrimitive(final Object value, final PrimitiveObjectInspector i
                 break;
             case TIMESTAMP:
                 Timestamp ts = ((TimestampObjectInspector) inspector).getPrimitiveJavaObject(value);
-                recordConsumer.addBinary(NanoTimeUtils.getNanoTime(ts, false).toBinary());
+                final NanoTime nanoTime = NanoTimeUtils.getNanoTime(ts, false);
+                ByteBuffer buffer = ByteBuffer.allocate(12);
+                buffer.order(ByteOrder.LITTLE_ENDIAN);
+                buffer.putLong(nanoTime.getTimeOfDayNanos());
+                buffer.putInt(nanoTime.getJulianDay());
+                buffer.flip();
+                recordConsumer.addBinary(Binary.fromByteBuffer(buffer));
```

Comment:
Why don't we use `nanoTime.toBinary()` here?

Reply:
Done.